core/modules/config/tests/config.test

```diff
@@ -158,6 +158,27 @@ class ConfigurationTest extends BackdropWebTestCase {
     $this->assertEqual(FALSE, $config_data->isOverridden('foo.key1'), 'Original value no longer overridden.');
     $this->assertEqual($config_data->get('foo.key1'), 'chocolate', 'Original value has been updated to include overridden value.');
   }
+
+  /**
+   * Tests that config directories are protected by .htaccess files.
+   */
+  public function testConfigHtaccess() {
+    // Check that the .htacess file is present.
+    $active_dir = config_get_config_directory();
+    $this->assertTrue(is_file($active_dir . '/.htaccess'), "Successfully located the .htaccess file in the 'active' config directory.");
+
+    // Verify contents of .htaccess file.
+    $file = file_get_contents($active_dir . '/.htaccess');
+    $this->assertEqual($file, file_htaccess_lines(FALSE), 'The .htaccess file contains the proper content.');
+
+    // Remove .htaccess file to then test that it gets re-created.
+    @backdrop_unlink($active_dir . '/.htaccess');
+    $this->assertFalse(is_file($active_dir . '/.htaccess'), "Successfully removed the .htaccess file in the 'active' config directory.");
+
+    // Do a normal web request to confirm that the file is re-created.
+    $this->backdropGet('admin/config/reports/status');
```

Comment:
That should be 'admin/reports/status'.

Additionally you need to create a user with permission to access that page.
```
    $admin_user = $this->backdropCreateUser(array(
      'access administration pages',
      'access site reports',
    ));
    $this->backdropLogin($admin_user);
    $this->backdropGet('admin/reports/status');
```